main.go

```diff
@@ -194,9 +193,10 @@ func hasBase64Prefix(p *edwards25519.Point, prefix []byte) bool {
 }
 
 func randUint64() uint64 {
-	r, err := rand.Int(rand.Reader, new(big.Int).SetUint64(math.MaxUint64))
+	var num uint64
+	err := binary.Read(rand.Reader, binary.LittleEndian, &num)
```

Comment:
This will not improve performance as it is called once per available CPU, i.e. only several times for the whole run.
But it makes code a tiny bit simpler (and I'd use binary.NativeEndian here).